lib/util/merge-confidence/index.spec.ts

```diff
@@ -43,157 +49,329 @@ describe('util/merge-confidence/index', () => {
     });
   });
 
-  describe('getMergeConfidenceLevel()', () => {
+  describe('API calling functions', () => {
+    const envOrg: NodeJS.ProcessEnv = process.env;
+    const hostRule: HostRule = {
+      hostType: 'merge-confidence',
+      token: 'some-token',
+    };
+
     beforeEach(() => {
-      hostRules.clear();
+      jest.resetAllMocks();
+      process.env = {
+        ...envOrg,
+        RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL: apiBaseUrl,
+      };
```

Comment:
as told before, don't override full env, just set the single env value. jest will isolate different test files, so no risk to influence tests in other files

Reply:
This change doesn't work:

```
diff --git a/lib/util/merge-confidence/index.spec.ts b/lib/util/merge-confidence/index.spec.ts
index f3abfd4ed..f7585945d 100644
--- a/lib/util/merge-confidence/index.spec.ts
+++ b/lib/util/merge-confidence/index.spec.ts
@@ -50,7 +50,6 @@ describe('util/merge-confidence/index', () => {
   });
 
   describe('API calling functions', () => {
-    const envOrg: NodeJS.ProcessEnv = process.env;
     const hostRule: HostRule = {
       hostType: 'merge-confidence',
       token: 'some-token',
@@ -58,17 +57,13 @@ describe('util/merge-confidence/index', () => {
 
     beforeEach(() => {
       jest.resetAllMocks();
-      process.env = {
-        ...envOrg,
-        RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL: apiBaseUrl,
-      };
+      process.env.RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL = apiBaseUrl;
       hostRules.add(hostRule);
       initConfig();
       memCache.reset();
     });
 
     afterEach(() => {
-      process.env = envOrg;
       hostRules.clear();
       resetConfig();
     });
```